app/module/drivers/kscan/kscan_gpio_direct.c

```diff
@@ -331,7 +352,9 @@ static const struct kscan_driver_api kscan_direct_api = {
                  "ZMK_KSCAN_DEBOUNCE_RELEASE_MS or debounce-release-ms is too large");             \
                                                                                                    \
     static struct kscan_gpio kscan_direct_inputs_##n[] = {                                         \
-        LISTIFY(INST_INPUTS_LEN(n), KSCAN_DIRECT_INPUT_CFG_INIT, (, ), n)};                        \
+        COND_CODE_1(DT_INST_NODE_HAS_PROP(n, input_gpios),                                         \
+                    (LISTIFY(INST_INPUTS_LEN(n), KSCAN_DIRECT_INPUT_CFG_INIT, (, ), n)),           \
```

Comment:
Renaming this to better differentiate it from the input-keys version could make this easier to understand.
```suggestion
                    (LISTIFY(INST_INPUTS_LEN(n), KSCAN_DIRECT_GPIO_INPUT_CFG_INIT, (, ), n)),           \
```